frontend/packages/kubevirt-plugin/src/components/connected-vm-console/connected-vm-console.tsx

```diff
@@ -0,0 +1,134 @@
+import * as React from 'react';
+import { K8sResourceKind, PodKind } from '@console/internal/module/k8s';
+import { PodModel, ServiceModel } from '@console/internal/models';
+import { getVMStatus } from '../../statuses/vm/vm-status';
+import { VMImportKind } from '../../types/vm-import/ovirt/vm-import';
+import { V1alpha1DataVolume } from '../../types/vm/disk/V1alpha1DataVolume';
+import {
+  DataVolumeModel,
+  VirtualMachineImportModel,
+  VirtualMachineInstanceMigrationModel,
+  VirtualMachineInstanceModel,
+  VirtualMachineModel,
+} from '../../models';
+import { VMIKind, VMKind } from '../../types/vm';
+import { Firehose, FirehoseResult } from '@console/internal/components/utils';
+import { VMConsolesWrapper } from '../vms/vm-console';
+import { getLoadedData } from '../../utils';
+import { ConsoleEmptyState } from './vm-console-empty-state';
+import { ConsoleType } from '../../constants/vm/console-type';
+
+const ConnectedVMConsole: React.FC<ConnectedVMConsoleProps> = ({
+  type,
+  vm,
+  vmis,
+  pods,
+  migrations,
+  dataVolumes,
+  vmImports,
+}) => {
+  const loadedVM = getLoadedData(vm);
```

Comment:
`getLoadedData` returns null in case of an error? later down the tree theres a component for many cases:
`VMCannotBeStarted, VMIsStarting, VMIsDown`

Reply:
I dont understand what kind of handling you can have to differentiate between loaded, loading and error states when getLoadedData can two values - data or null. That's 3 states but only 2 values ?

Reply:
I thought `getVMStatus` handles it pretty nicely?

Reply:
Seems to me like you would end up with `<VMIsDown />` while loading/error. There's just no state for loading or error

Reply:
IMO it is not that important to handle this in VMConsolesTab.
It might be useful a bit in standaloneconsole. I don't mind either way. The user will probably visit the VM page when the console stops working (vm down) to debug it.